File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralItem/IfcStructuralMember/IfcStructuralCurveMember/IfcStructuralCurveMember.vb
Public Class IfcStructuralCurveMember Inherits IfcStructuralMember
    Public PredefinedType As IfcStructuralCurveMemberTypeEnum
    Public Axis As IfcDirection

    ' === EXTENDED BY ===
    ' IfcStructuralCurveMemberVarying

    ' === WHERE CLAUSES ===
    ' HasObjectType : (PredefinedType <> IfcStructuralCurveMemberTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralItem/IfcStructuralMember/IfcStructuralCurveMember/IfcStructuralCurveMember.java
public class IfcStructuralCurveMember extends IfcStructuralMember {
    public IfcStructuralCurveMemberTypeEnum PredefinedType;
    public IfcDirection Axis;

    // === EXTENDED BY ===
    // IfcStructuralCurveMemberVarying

    // === WHERE CLAUSES ===
    // HasObjectType : (PredefinedType <> IfcStructuralCurveMemberTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
}
